### ISSUE TITLE
MD051 option for space replacement character

### ISSUE THREAD

**Issue by nschonni** [DavidAnson/markdownlint#606]
Currently the space is swapped by `-`, I think there have been one or to other parsers that flagged that they use `_` instead, so it might help to keep the default, but allow overriding this.

This is the current replacement, that would need to take the option:
```js
.replace(/ /gu, "-")
```

I was wondering if some larger configurability might be needed for the different anchor link patterns, but maybe this would be enough

**Comment by DavidAnson** [comment]
Update: A request for a different algorithm entirely: https://github.com/DavidAnson/markdownlint/issues/570

**Comment by nschonni** [comment]
Yes, this is a lighter version compared to that issue. The main use case I have for this is coming from the mdn/content repo. Since they migrated from a Wiki format to markdown, they maintained the `_` style replacement

**Comment by connorads** [comment]
I've also faced similar issue with `MD051` and Backstage TechDocs which replaces ` - ` (space dash space) with `-` (dash)

Interesting that MDN have chosen to disable `MD051` because of different algorithm in their docs system yari
https://github.com/mdn/content/blob/main/.markdownlint-cli2.jsonc#L117-L118

Would be great to be able to configure this somehow as `MD051` does great work to keep them links working!

_In the meantime I've been looking for a way to stop people from using `-` in headings._
